web3/_utils/abi.py

```diff
@@ -256,113 +275,41 @@ class BytesEncoder(AcceptsHexStrEncoder):
     is_strict = False
 
 
-class ByteStringEncoder(AcceptsHexStrEncoder):
-    subencoder_cls = encoding.ByteStringEncoder
-    is_strict = False
-
-
-class StrictByteStringEncoder(AcceptsHexStrEncoder):
-    subencoder_cls = encoding.ByteStringEncoder
+class ExactLengthBytesEncoder(BytesEncoder):
     is_strict = True
 
-
-class ExactLengthBytesEncoder(encoding.BaseEncoder):
-    # TODO: move this to eth-abi once the api is stabilized
-    is_big_endian = False
-    value_bit_size = None
-    data_byte_size = None
-
     def validate(self) -> None:
         super().validate()
-
-        if self.value_bit_size is None:
-            raise ValueError("`value_bit_size` may not be none")
         if self.data_byte_size is None:
             raise ValueError("`data_byte_size` may not be none")
-        if self.encode_fn is None:
-            raise ValueError("`encode_fn` may not be none")
         if self.is_big_endian is None:
             raise ValueError("`is_big_endian` may not be none")
 
-        if self.value_bit_size % 8 != 0:
```

Comment:
Yeah... `value_bit_size` was defined but not actually being used anywhere. This wasn't doing anything as it was. Instead, every validation step was checking against the `data_byte_size` property.

Reply:
Hmm I actually don't see why we shouldn't put the bit of validation that's here back. I think it's hard to pass in invalid numbers since it is being instantiated using `from_type_str` every time (which `does data_byte_size * 8` every time... but in case the class is initialized in an unexpected way we'd at least have that check in there.

Reply:
Okay, cool. Yeah, probably not a huge problem since we didn't even have tests that were failing.

Reply:
👍🏼 I did add a test for the case I was thinking of, just to be safe.